eth/stagedsync/stage_mining_exec.go

```diff
@@ -147,6 +148,12 @@ func SpawnMiningExecStage(s *StageState, tx kv.RwTx, cfg MiningExecCfg, quit <-c
 					break
 				}
 			}
+
+			parent := rawdb.ReadHeaderByNumber(tx, executionAt)
```

Comment:
`current` (from what I understand) holds the next proposed block, but here we want to measure the delay from the `timestamp` in the previous block header. It seems like we do need the additional DB read here.

Reply:
@shohamc1 ok, in this case we can do something else: have a look at `stage_mining_create_block` - it already reads the parent using `ReadHeaderByNumber` since it needs that information. So you can "memorise" this information in `MiningState.MiningBlock.Parent`